### DISCUSSION TITLE
ZodSchema for the openapi document

### DISCUSSION THREAD

**Discussion by RonB** [samchungy/zod-openapi#376] [Q&A] [answered]
Hi,

As i would like to include the response of your great work as an api endpoint itself, i would need a Zod schema for the entire OpenAPiDocument.

```json
paths: {
    '/api/openapi': {
      get: {
        requestParams: { query: z.object({format: z.enum(['json','yaml').default('json').describe("Format to be returned") },
        responses: {
          '200': {
            description: '200 OK',
            content: {
              'application/json': { schema: ZodOpenApiDocumentSchema },
            },
          },
        },
      },
    }
  }  
```

Would that be incredibly difficult?

Thanks in advance and thank you for this awesome library.

Regards from the Netherlands, Ronald

**Comment by samchungy** [comment, marked as answer]
You're probably better off referencing the spec itself: https://github.com/OAI/OpenAPI-Specification/blob/main/schemas/v3.1/schema.yaml

**Reply by RonB** [reply]
fair enough... I have everything working now with dynamic api paths based on predefined descriptions. Class methods are decorated with a "registry" which i use for generating the swagger.json using your awesome library.

Another usage for me is the transformation of my zod schemas to my fastapi hub (python) that uses pydantic schemas. The datamodel-codegen tool can absorb the open api document and transfer that directly to pydantic classes. This is a huge win  otherwise i had to keep over 50 schema's in sync.
